local ReplicatedStorage = game:GetService("ReplicatedStorage")

local Base36 = require(ReplicatedStorage.Modules.Shared.Base36)
local TableUtil = require(ReplicatedStorage.Packages.TableUtil)

local sponsoredCodes = {
	"10",
	"60",
	"11",
	"12",
	"13",
	"14",
	"15",
	"16",
	"17",
	"18",
	"1A",
	"1B",
	"1C",
	"1D",
	"1F",
	"1G",
	"1H",
	"1I",
	"1J",
	"1K",
	"1L",
	"1M",
	"1N",
	"1O",
	"1P",
	"1Q",
	"1R",
	"1S",
	"1T",
	"1U",
	"1V",
	"1W",
	"1X",
	"1Y",
	"2",
	"20",
	"21",
	"22",
	"23",
	"24",
	"25",
	"26",
	"27",
	"28",
	"29",
	"2A",
	"2B",
	"2C",
	"2D",
	"2E",
	"2F",
	"2G",
	"2H",
	"2I",
	"2J",
	"2K",
	"2L",
	"2M",
	"2N",
	"2O",
	"2P",
	"2Q",
	"2R",
	"2S",
	"2T",
	"2U",
	"2V",
	"2W",
	"2X",
	"2Y",
	"2Z",
	"30",
	"31",
	"32",
	"33",
	"34",
	"35",
	"36",
	"37",
	"38",
	"39",
	"3A",
	"3B",
	"3C",
	"3D",
	"3E",
	"3F",
	"3G",
	"3H",
	"3I",
	"3J",
	"3K",
	"3L",
	"3M",
	"3N",
	"3P",
	"3Q",
	"3R",
	"3S",
	"3T",
	"3U",
	"3V",
	"3W",
	"3X",
	"3Y",
	"7",
	"9",
	"A",
	"B",
	"C",
	"D",
	"E",
	"F",
	"G",
	"H",
	"I",
	"J",
	"K",
	"L",
	"M",
	"N",
	"O",
	"P",
	"Q",
	"R",
	"S",
	"T",
	"U",
	"V",
	"X",
	"Y",
	"Z",
}

local numericCodes = TableUtil.Map(sponsoredCodes, Base36.Decode)

return numericCodes
